src/test/java/sirius/kernel/commons/ValueSpec.groovy

```diff
@@ -168,6 +168,77 @@ class ValueSpec extends BaseSpecification {
         Value.of(Amount.of(0.00001)).getAmount() == Amount.of(0.00001)
     }
 
+    def "Casting to integers works"() {
+        expect:
+        Value.of(input).asInt(1234) == output
+        where:
+        input             | output
+        99999             | 99999
+        99999l            | 99999
+        99999.0           | 99999
+        new Double(99999) | 99999
+        Amount.of(99999)  | 99999
+        "99999"           | 99999
+        "Keine Zahl"      | 1234
+    }
+
+    def "Casting to integer instances works"() {
+        expect:
+        Value.of(input).getInteger() == output
+        where:
+        input             | output
+        99999             | new Integer(99999)
+        99999l            | new Integer(99999)
+        99999.0           | new Integer(99999)
+        new Double(99999) | new Integer(99999)
+        Amount.of(99999)  | new Integer(99999)
+        "99999"           | new Integer(99999)
+        "Keine Zahl"      | null
+    }
+
+    def "Casting to longs works"() {
+        expect:
+        Value.of(input).asLong(1234) == output
+        where:
+        input             | output
+        99999             | 99999l
+        99999l            | 99999l
+        99999.0           | 99999l
+        new Double(99999) | 99999l
+        Amount.of(99999)  | 99999l
+        "99999"           | 99999l
+        "Keine Zahl"      | 1234l
+    }
+
+    def "Casting to long instances works"() {
+        expect:
+        Value.of(input).getLong() == output
+        where:
+        input             | output
+        99999             | new Long(99999l)
+        99999l            | new Long(99999l)
+        99999.0           | new Long(99999l)
+        new Double(99999) | new Long(99999l)
+        Amount.of(99999)  | new Long(99999l)
+        "99999"           | new Long(99999l)
+        "Keine Zahl"      | null
+    }
+
+    def "Casting to doubles works"() {
+        expect:
+        Value.of(input).asDouble(1234.56) == output
```

Comment:
Is this a valid double comparison in Groovy? There should be an ε

Reply:
Doubles.EPSILON?